RosNet/RosReader/RosBagReader.cs

```diff
@@ -61,8 +83,257 @@ public static RosBag Read(string path)
             }
             else
             {
-                throw new FileNotFoundException("File does not exist");
+                throw new FileNotFoundException($"File with path {path} does not exist");
+            }
+        }
+
+        /// <summary>
+        /// Reads a connection and sets the data
+        /// </summary>
+        private static void SetConnectionData(BinaryReader reader, ref Connection connection)
```

Comment:
You already have `ReadHeader`! Why not just use that and iterate over the headers you read?

Reply:
Why not just make `Connection` take all the data it needs in it's constructor? By not doing so you allow the possibility to create _invalid state_ (e.g. values can be null when they shouldn't be), when the possibility of creating that at all can be completely avoided. This becomes especially clear with e.g. the nullability-warnings since `Nullable` was enabled in #72

Reply:
Added issue #76